solutions/standard/main.tf

```diff
@@ -83,7 +111,7 @@ module "secrets_manager" {
   kms_encryption_enabled            = true
   existing_kms_instance_guid        = local.existing_kms_guid
   kms_key_crn                       = local.kms_key_crn
-  skip_kms_iam_authorization_policy = var.skip_kms_iam_authorization_policy
+  skip_kms_iam_authorization_policy = var.skip_kms_iam_authorization_policy || var.ibmcloud_kms_api_key != null
```

Comment:
This logic may be confusing. As a consumer, I may want to skip_kms_iam_authorization_policy = false to make sure one is created. I may then pass a key to use to create it.
Should this be flipped to true IF apply_cross_account_auth_policy is true? So it always skipped IFF one was just created?

Reply:
Using only `apply_cross_account_auth_policy` might give the wrong impression that we either create an inter-account or intra-account auth policy. However, there might be cases where we create neither. Therefore, updating this as follows:

```
skip_kms_iam_authorization_policy = var.skip_kms_iam_authorization_policy || local.create_cross_account_auth_policy```